pytorch_lightning/utilities/distributed.py

```diff
@@ -376,3 +376,28 @@ def init_dist_connection(
             f"All distributed processes registered. Starting with {world_size} processes\n"
             f"{'-' * 100}\n"
         )
+
+
+def _collect_states_on_rank_zero(state: Dict[str, Any], device: torch.device) -> Optional[Dict[int, Any]]:
+    """This distributed utility collects dictionary state across all processes.
+
+    Args:
+        state: Dictionary contain the state of the current process
+        device: Current process device.
+
+    Returns:
+        states: On global rank 0, a dictionary where the primary keys are
+            the process rank and the values their associated states. Otherwise, returns None.
+    """
+    if not distributed_available():
+        return {0: state}
+    states = {}
+    current_rank = torch.distributed.get_rank()
+    for rank in range(1, torch.distributed.get_world_size()):
+        objects = [state if current_rank == rank else None]  # type: ignore
```

Comment:
don't need this anymore :)

```suggestion
        objects = [state if current_rank == rank else None]
```